src/language_features/document_symbol.rs

```diff
@@ -23,6 +23,125 @@
 use unicode_width::UnicodeWidthStr;
 use url::Url;
 
+pub fn breadcrumbs(meta: EditorMeta, editor_params: EditorParams, ctx: &mut Context) {
+    let eligible_servers: Vec<_> = ctx
+        .language_servers
+        .iter()
+        .filter(|srv| attempt_server_capability(*srv, &meta, CAPABILITY_DOCUMENT_SYMBOL))
+        .collect();
+    let req_params = eligible_servers
+        .into_iter()
+        .map(|(server_name, _)| {
+            (
+                server_name.clone(),
+                vec![DocumentSymbolParams {
+                    text_document: TextDocumentIdentifier {
+                        uri: Url::from_file_path(&meta.buffile).unwrap(),
+                    },
+                    partial_result_params: Default::default(),
+                    work_done_progress_params: Default::default(),
+                }],
+            )
+        })
+        .collect();
+    let params = BreadcrumbsParams::deserialize(editor_params).unwrap();
+    ctx.call::<DocumentSymbolRequest, _>(
+        meta,
+        RequestParams::Each(req_params),
+        move |ctx: &mut Context, meta, results| {
+            match results.into_iter().find(|(_, v)| v.is_some()) {
+                Some((server_name, Some(DocumentSymbolResponse::Nested(symbols))))
+                    if !symbols.is_empty() =>
+                {
+                    let server = &ctx.language_servers[&server_name];
+                    let document = if let Some(document) = ctx.documents.get(&meta.buffile) {
+                        document
+                    } else {
+                        return;
```

Comment:
We can use `let-else` nowadays (`let Some(document) = ctx.documents.get(&meta.buffile) else { return }`).
Also this seems like a very exceptional case, I'll add a warning. No need to forward it to the user I guess since this is always called from a hook.

actually, clippy informed me that this is unused, so we can remove it altogether